src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -177,6 +206,25 @@ async def infer_python_dependencies_via_imports(
         if maybe_disambiguated:
             merged_result.add(maybe_disambiguated)
 
+        if not owners.unambiguous and imp.split(".")[0] not in DEFAULT_UNOWNED_DEPENDENCIES:
+            unowned_imports.add(imp)
+
+    unowned_dependency_behavior = python_infer_subsystem.unowned_dependency_behavior
+    if unowned_imports and unowned_dependency_behavior is not UnownedDependencyUsage.DoNothing:
+        raise_error = unowned_dependency_behavior is UnownedDependencyUsage.RaiseError
+        log = logger.error if raise_error else logger.warning
+        log(
+            f"The following imports in {address} have no owners:\n\n{bullet_list(unowned_imports)}\n\n"
+            "If you are using [python].requirement_constraints, consider adding the relevant package.\n"
+            "Otherwise consider specifying a python_requirement target as a dependency.\n"
```

Comment:
```suggestion
            "If you are using a requirements file, consider adding the relevant package.\n"
            "Otherwise consider declaring a `python_requirement_library` target, which can then be inferred.\n"
```

To avoid the warning, you'll need a `python_requirements_library`, one way or another: https://www.pantsbuild.org/docs/reference-python_requirement_library ... but they are generated by the `python_requirements` macro.

This also assumes that the library is thirdparty... usually a safe assumption, but maybe not in a brand new repo? Could say something like:
"If you are expecting this import to be provided by your own firstparty code, ensure that it is contained within a [source root](https://www.pantsbuild.org/docs/source-roots)."